components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/EndpointListenerLoader.java

```diff
@@ -97,10 +96,9 @@ public static void loadListeners() {
         }
 
         int internalInboundPort = HTTPEndpointManager.getInstance().getInternalInboundPort();
-        if (internalInboundPort != -1) {
-            HTTPEndpointManager.getInstance().startListener(
-                    internalInboundPort, InboundHttpConstants.INTERNAL_INBOUND_ENDPOINT_NAME, null);
-
+        if (internalInboundPort != -1 && HTTPEndpointManager.getInstance().isAnyInternalApiEnabled()) {
```

Comment:
Can the internalInboundPort be -1? We need to default it to a valid port.

Reply:
yes this can be, since we are returning -1 in many places

https://github.com/wso2/carbon-mediation/blob/master/components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/internal/http/api/ConfigurationLoader.java#L145

https://github.com/wso2/carbon-mediation/blob/master/components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/internal/http/api/ConfigurationLoader.java#L141

and setting default port after only reading the synapse properties file 

https://github.com/wso2/carbon-mediation/blob/master/components/inbound-endpoints/org.wso2.carbon.inbound.endpoint/src/main/java/org/wso2/carbon/inbound/endpoint/internal/http/api/ConfigurationLoader.java#L137